packages/client3/components/design-system/typography/heading.tsx

```diff
@@ -2,30 +2,31 @@ import clsx from "clsx";
 import { HTMLAttributes, ComponentType } from "react";
 
 export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
-  level: 1 | 2 | 3 | 4 | 5;
+  level: 1 | 2 | 3 | 4 | 5 | 6;
+  /** This option is only valid for level 5 and 6 */
+  medium?: boolean;
   as?: string | ComponentType<{ className: string }>;
 }
 
-const levelToFontSize = {
-  5: "text-3xl",
-  4: "text-4xl",
-  3: "text-5xl",
-  2: "text-6xl",
-  1: "text-5xl md:text-7xl",
-};
-
-export function Heading({ level, as, className, ...rest }: HeadingProps) {
+export function Heading({
+  level,
+  medium = false,
+  as,
+  className,
+  ...rest
+}: HeadingProps) {
   const Component = as ?? `h${level}`;
 
-  const fontSizeClass = levelToFontSize[level];
-
   return (
     <Component
       className={clsx(
-        level === 1 || level === 2 ? "font-serif" : "font-sans",
-        "tracking-[0.02rem] text-sand-800",
-        level === 1 ? "font-bold" : "font-semibold",
-        fontSizeClass,
+        level === 1
+          ? "font-bold"
+          : level < 5
+          ? "font-semibold"
+          : medium
+          ? "font-medium"
+          : "font-semibold",
```

Comment:
I would suggest using objects when there are multiple ternary operations involved.
Something like this,
`clsx("font-semibold",{"font-semibold":level < 5,"font-bold":level === 1,"font-medium":medium},className)`

Thoughts?

Reply:
I'm not quite familiar with this, but I've tested, and doing this way results in wrong results with the fonts. such as:
`font-semibold font-semibold` or `font-semibold font-bold`